subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt

```diff
@@ -305,14 +328,28 @@ class SubscriptionFeedbackViewModel @Inject constructor(
                         isForward = false,
                     )
 
-                    is FeedbackSubCategory -> ViewState(
-                        feedbackMetadata = currentFeedbackMetadata.copy(
-                            subCategory = null,
-                        ),
-                        currentFragmentState = newState,
-                        previousFragmentState = FeedbackCategory(currentFeedbackMetadata.reportType?.asTitle() ?: -1),
-                        isForward = false,
-                    )
+                    is FeedbackSubCategory -> {
+                        val previousState =
+                            if (currentFeedbackMetadata.reportType == REPORT_PROBLEM &&
+                                (
+                                    currentFeedbackMetadata.source == SUBSCRIPTION_SETTINGS ||
+                                        currentFeedbackMetadata.source == VPN_MANAGEMENT ||
+                                        currentFeedbackMetadata.source == VPN_EXCLUDED_APPS
+                                    )
+                            ) {
```

Comment:
Might be nice to extract to courtesy (possibly extension) property or function for readability. e.g.

```suggestion
                            if (currentFeedbackMetadata.isReportAboutSubscriptionOrVpn) {
```